lmnet/lmnet/datasets/dataset_iterator.py

```diff
@@ -79,12 +79,14 @@ def _xorshift32(r):
 
 
 class _MultiProcessDatasetPrefetchThread(threading.Thread):
-    def __init__(self, dataset, result_queue, seed):
+    def __init__(self, dataset, result_queue, seed, queue_size=200):
         super().__init__()
         # TODO(tokunaga): the number of processes should be configurable
         self.seed = seed + 1  # seed must not be 0
         self.support_getitem = hasattr(dataset, "__getitem__")
-        self.pool = Pool(processes=8, initializer=_prefetch_setup,
+        self.num_proc = max(cpu_count() // 2, 1)
```

Comment:
Why do you use half of cpu count ? It is better to custom this number by user.

Reply:
@ruimashita San: Fixed in the next commit.
I've created a config variable DATASET.NUM_PARALLEL_PROCESS and use this for number of processes. The default is half of CPU cores. I choose this value considering it is always good to have some cpu resources available for system processes to avoid system freezing.